File: ruby/hamming/hamming.rb
class Hamming
  def self.compute(strand_a, strand_b)
    raise ArgumentError unless strand_a.length == strand_b.length

    strand_a.each_char.with_index.count do |char, idx|
      char != strand_b[idx]
    end
  end
end

module BookKeeping
  VERSION = 3
end

File: java/hamming/src/main/java/Hamming.java
public class Hamming {
    public static int compute(String strandA, String strandB) {
        if (strandA.length() != strandB.length()) {
            throw new IllegalArgumentException();
        }

        int distance = 0;
        for (int i = 0; i < strandA.length(); i++) {
            if (strandA.charAt(i) != strandB.charAt(i)) {
                distance++;
            }
        }

        return distance;
    }
}
